### ISSUE TITLE
Dependent Field not works for relation manager on the record creation form

### ISSUE THREAD

**Issue by hadolen** [laravel-filament/filament#240]
Conditions are not triggered on the record creation form.

**Comment by danharrin** [comment]
Hi @hadolen, could you please provide a form code example where this doesn't work? Thanks

**Comment by hadolen** [comment]
This code doesn't work

```
<?php

namespace App\Filament\Resources\ProductResource\RelationManagers;

use Filament\Resources\Forms\Components;
use Filament\Resources\Forms\Form;
use Filament\Resources\RelationManager;
use Filament\Resources\Tables\Columns;
use Filament\Resources\Tables\Filter;
use Filament\Resources\Tables\Table;

class OptionsRelationManager extends RelationManager
{
    public static $primaryColumn = '';

    public static $relationship = 'options';

    public static function form(Form $form)
    {
        return $form
            ->schema([
                         
                Components\BelongsToSelect::make('option_id')->relationship('option', 'title')->preload()->dependable(),
                
                Components\BelongsToSelect::make('option_value_id')->preload()->relationship('option_value', 'title')->when(
                    function ($record) {            
                        return true;
                    },
                    
                    function ($field) {                    
                        $field->relationship('option_value', 'title', function ($query) use ($field) {
                            return $query->where('option_id', $field->getRecord() == null ? 0 : $field->getRecord()->option_id);
                        });
                    }
                    
                ),
            ]);
    }

    public static function table(Table $table)
    {
        return $table
            ->columns([
                //
            ])
            ->filters([
                //
            ]);
    }
}

```

```
<?php

namespace App\Filament\Resources;

use App\Filament\Resources\ProductResource\Pages;
use App\Filament\Resources\ProductResource\RelationManagers;
use App\Filament\Roles;
use Filament\Resources\Forms\Components;
use Filament\Resources\Forms\Form;
use Filament\Resources\Resource;
use Filament\Resources\Tables\Columns;
use Filament\Resources\Tables\Filter;
use Filament\Resources\Tables\Table;

class ProductResource extends Resource
{
    public static $icon = 'heroicon-o-collection';

    public static function form(Form $form)
    {
        return $form
            ->schema([
                Components\TextInput::make('title')->autofocus()->required(),
                Components\RichEditor::make('description')->required(),
                Components\TextInput::make('price')->numeric()->min(0)->extraAttributes(['step' => 0.10])->required(),
            ]);
    }

    public static function table(Table $table)
    {
        return $table
            ->columns([
                Columns\Text::make('title')->sortable()->searchable()->primary(),
                Columns\Text::make('price')->sortable(),
            ])
            ->filters([
                //
            ]);
    }

    public static function relations()
    {
        return [
            RelationManagers\OptionsRelationManager::class,
        ];
    }

    public static function routes()
    {
        return [
            Pages\ListProducts::routeTo('/', 'index'),
            Pages\CreateProduct::routeTo('/create', 'create'),
            Pages\EditProduct::routeTo('/{record}/edit', 'edit'),
        ];
    }
}

```

But similar code outside relation manager work fine.

```
<?php

namespace App\Filament\Resources;

use App\Filament\Resources\ProductOptionResource\Pages;
use App\Filament\Resources\ProductOptionResource\RelationManagers;
use App\Filament\Roles;
use Filament\Resources\Forms\Components;
use Filament\Resources\Forms\Form;
use Filament\Resources\Resource;
use Filament\Resources\Tables\Columns;
use Filament\Resources\Tables\Filter;
use Filament\Resources\Tables\Table;

class ProductOptionResource extends Resource
{
    public static $icon = 'heroicon-o-collection';

    public static function form(Form $form)
    {
        return $form
            ->schema([
                Components\BelongsToSelect::make('product_id')->relationship('product', 'title')->preload(),               
                
                Components\BelongsToSelect::make('option_id')->relationship('option', 'title')->preload()->dependable(),
                
                Components\BelongsToSelect::make('option_value_id')->preload()->relationship('option_value', 'title')->when(
                    function ($record) {            
                        return true;
                    },
                    
                    function ($field) {                    
                        $field->relationship('option_value', 'title', function ($query) use ($field) {
                            return $query->where('option_id', $field->getRecord() == null ? 0 : $field->getRecord()->option_id);
                        });
                    }
                    
                ),
                
            ]);
    }

    public static function table(Table $table)
    {
        return $table
            ->columns([
                //
            ])
            ->filters([
                //
            ]);
    }

    public static function relations()
    {
        return [
            //
        ];
    }

    public static function routes()
    {
        return [
            Pages\ListProductOptions::routeTo('/', 'index'),
            Pages\CreateProductOption::routeTo('/create', 'create'),
            Pages\EditProductOption::routeTo('/{record}/edit', 'edit'),
        ];
    }
}

```

**Comment by danharrin** [comment]
Fixed by #250.
